### PR TITLE
Read layout information using libxkbcommon

### PR DESCRIPTION
**Pull request by tari01**

### CODE REVIEW

File: .build.yml [hunk 1]
```diff
@@ -83,6 +101,19 @@ before_scripts:
...
+  -     if [ ! -d libxkbcommon-build ]; then
+  -         git clone --depth 1  https://github.com/xkbcommon/libxkbcommon.git libxkbcommon-build
+  -     fi
+  -     cd libxkbcommon-build
+  -     meson setup build -Dprefix=/usr -Dlibdir=/usr/lib/x86_64-linux-gnu
+  -     ninja -C build
+  -     ninja -C build install
+  -     cd -
+  -     rm -Rf libxkbcommon-build/
+  - fi
```
Comment (sunweaver) [inline .build.yml:116]: I see no reason for building libxkbcommon from source here. Can't we use libxkbcommon in the various versions found in the distros / distro versions that we support?

File: src/keyboard-x11.c [hunk 2]
```diff
@@ -292,6 +255,53 @@ Keyboard* keyboard_new()
...
+        g_warning("Failed to include default paths");
+    }
+
+    if (!rxkb_context_parse(pContext, "evdev"))
+    {
+        g_warning("Failed to parse XKB descriptions");
+    }
+
+    struct rxkb_layout *pRxkbLayout = rxkb_layout_first(pContext);
+    g_assert(pRxkbLayout);
```
Comment (sunweaver) [inline src/keyboard-x11.c:275]: This g_assert call should either be dropped or accompanied by a -DG_DISABLE_ASSERT in the compiler options.

**Review by sunweaver — CHANGES_REQUESTED** [review]
See inline comments...

File: .build.yml (hunk 1 above)
Comment (tari01) [reply .build.yml:116]: Ubuntu 20.04 does not have `libxkbregistry`.

File: src/keyboard-x11.c (hunk 2 above)
Comment (tari01) [reply src/keyboard-x11.c]: Done.

**Review by sunweaver — APPROVED** [review]
ACK, let's see where this gets.

**Pull request merged**